format/nexus.py

```diff
@@ -535,26 +538,41 @@ class BeamFactory(object):
     A class to create a beam model from NXmx stuff
     """
 
-    def __init__(self, obj, index=None):
+    def __init__(self, obj):
```

Comment:
This changes the constructor signature and class behaviour on a public class. Can the fixes be backported as a minimal changeset that retain the current class behaviour? Alternatively I suggest a second class is introduced with the changed behaviour.